File: DSA_python/DSA01004 SINH TỔ HỢP.py
def c(n, k):
    total = 1
    for i in range(1, k+1):
        total *= (n-i+1)
        total /= i
    return int(total)

def update(A, n, k):
    i = k-1
    while A[i] == n-k+i+1:
        i -= 1
    if i == -1:
        return

    A[i] += 1
    for j in range(i+1, k):
        A[j] = A[i]+j-i

    return A

def submit():
    n, k = map(int, input().split())
    A = []
    for i in range(1, k+1):
        A.append(i)

    for i in range(c(n,k)):
        for j in A:
            print(j, end = "")
        print(end = " ")
        A = update(A, n, k)

    print()

test_num = int(input())
for i in range(test_num):
    submit()

File: DSA/DSA01004 SINH TỔ HỢP.cpp
#include <bits/stdc++.h>
using namespace std;

void update(vector<int> &A, int n, int k)
{
    int i = k-1;
    while (A[i] == n-k+i+1)
    {
        i--;
    }
    
    if (i == -1)
    {
        return;
    }
    
    A[i]++;
    for (int j = i+1; j < k; j++)
    {
        A[j] = A[i]+j-i;
    }
}

long long c(int n, int k)
{
    long long total = 1;
    for (int i = 1; i <= k; i++)
    {
        total *= n-i+1;
        total /= i;
    }
    return total;
}

void submit()
{
    int n, k;
    cin >> n >> k;
    
    vector<int> A(k);
    for (int i = 0; i < k; i++)
    {
        A[i] = i+1;
    }
    
    for (long long i = 0; i < c(n, k); i++)
    {
        for (auto &&j : A)
        {
            cout << j;
        }
        cout << " ";
        update(A, n, k);
    }
    
    cout << endl;
}   

int main()
{
    ios::sync_with_stdio(false);                                    // Speed up input/output
    cin.tie(nullptr);
    cout.tie(nullptr);

    int so_bo_test;
    cin >> so_bo_test;
    cin.ignore();

    while (so_bo_test--)
    {
        submit();
    }

    return 0;
}